### DISCUSSION TITLE
sing-box的geosite和ip文件，如何更新？

### DISCUSSION THREAD

**Discussion by Travel2Here** [Openwrt-Passwall/openwrt-passwall2#420] [General]
原本xray的这两个文件在规则更新模块，可以设自动和手动即时更新。现在singbox的这个配置在高级设置中，且没有发现如何触发更新。如何是好？（目前人肉覆盖目录文件）

**Comment by rianjskis** [comment]
定时任务覆盖

**Comment by 2eronia** [comment]
自己搜搜怎么写 shell 通过 wget 或者 curl 来下载、覆盖

**Comment by CharlesLocke** [comment]
![2024-02-23_14-43-58](https://github.com/xiaorouji/openwrt-passwall2/assets/22140090/9dc8e2bb-c5f9-4c6d-a239-8cfb4f92c9ed)

**Comment by VergilGao** [comment]
wget 版本的：
```
0 3 * * * wget -O /tmp/geoip.db https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geoip.db && [ $(find /tmp/geoip.db -printf "%s") -gt 1048576 ] && mv /tmp/geoip.db /usr/share/singbox/geoip.db
0 3 * * * wget -O /tmp/geosite.db https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geosite.db && [ $(find /tmp/geosite.db -printf "%s") -gt 1048576 ] && mv /tmp/geoip.db /usr/share/singbox/geosite.db
```

这个命令的作用是将geoip和geosite下载到 /tmp 然后判断一下下载下来的文件大小是否大于 1M 如果大于 1M 则将他复制到 /usr/share/singbox

如果有其他办法来判断下载的 db 文件的合法性请务必告诉我。

**Reply by chshouyu** [reply]
更新之后不需要重启 passwall2 吗？

**Reply by fyngd** [reply]
> wget 版本的：
> 
> ```
> 0 3 * * * wget -O /tmp/geoip.db https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geoip.db && [ $(find /tmp/geoip.db -printf "%s") -gt 1048576 ] && mv /tmp/geoip.db /usr/share/singbox/geoip.db
> 0 3 * * * wget -O /tmp/geosite.db https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geosite.db && [ $(find /tmp/geosite.db -printf "%s") -gt 1048576 ] && mv /tmp/geoip.db /usr/share/singbox/geosite.db
> ```
> 
> 这个命令的作用是将geoip和geosite下载到 /tmp 然后判断一下下载下来的文件大小是否大于 1M 如果大于 1M 则将他复制到 /usr/share/singbox
> 
> 如果有其他办法来判断下载的 db 文件的合法性请务必告诉我。

楼主，你这个好像不会自动替换旧版啊。
